packages/flutter/lib/src/painting/image_stream.dart

```diff
@@ -42,19 +44,23 @@ class ImageInfo {
   /// (e.g. in the arguments given to [Canvas.drawImage]).
   final double scale;
 
+  /// A string used for debugging purpopses to identify the source of this image.
+  final String debugLabel;
+
   @override
-  String toString() => '$image @ ${debugFormatDouble(scale)}x';
+  String toString() => '${debugLabel != null ? '$debugLabel ' : ''}$image @ ${debugFormatDouble(scale)}x';
```

Comment:
Nit: if you'd like to make sure you don't accidentally add an extra space you can write this code like this
```dart
[ if (debugLabel != null) debugLabel, image, '@${debugFormatDouble(scale)}'].join(' ')]
```
not sure it is better.